cmd/common/utils/parser-tableformatter.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/columns"
+	"github.com/kinvolk/inspektor-gadget/pkg/columns/formatter/textcolumns"
+)
+
+const (
+	KubernetesTag       string = "kubernetes"
+	ContainerRuntimeTag string = "runtime"
+)
+
+var SupportedTags = map[string]struct{}{
+	KubernetesTag:       {},
+	ContainerRuntimeTag: {},
+}
+
+func WithMetadata[T any](tag string) func(*GadgetParser[T]) error {
+	return func(parser *GadgetParser[T]) error {
+		if _, ok := SupportedTags[tag]; !ok {
+			return fmt.Errorf("unknown value for metadata: %q", tag)
+		}
+
+		// The resulting set of columns will be the gadget-specific columns plus
+		// the ones with the requested tag.
+		parser.columnsFilter = columns.Or(columns.WithNoTags(), columns.WithTag(tag))
+
+		return nil
+	}
+}
+
+// GadgetParser is a parser helps printing the gadget output in columns using
+// the columns and tableformatter packages.
+type GadgetParser[T any] struct {
+	columns       *columns.Columns[T]
+	columnsFilter columns.ColumnFilter
+	formatter     *textcolumns.TextColumnsFormatter[T]
+}
+
+func NewGadgetParser[T any](outputConfig *OutputConfig, options ...func(*GadgetParser[T]) error) (*GadgetParser[T], error) {
+	cols, err := columns.NewColumns[T]()
+	if err != nil {
+		return nil, fmt.Errorf("creating columns: %w", err)
+	}
+
+	parser := &GadgetParser[T]{
+		columns: cols,
+
+		// By default, use only the columns with no specific tag. In other
+		// words, the gadget-specific columns. Notice the filter could be
+		// modified using WithCustomConfig.
+		columnsFilter: columns.WithNoTags(),
```

Comment:
I think it makes sense - if we ever run into a situation where this isn't enough, we could add a "gadget" tag.
For the interface, I think it is more common to call the function HasTags() (instead of HasNoTags()) and then negate that in WithNoTags().

Reply:
> For the interface, I think it is more common to call the function HasTags() (instead of HasNoTags()) and then negate that in WithNoTags().

Mmmm, but in WithNoTags(), we don't have the information about the "available" tags. So, which tags should I use to call HasTag()? Maybe I am missing something.

Reply:
I meant this interface here: https://github.com/kinvolk/inspektor-gadget/pull/934/files#diff-fe87426b97ca95399ca7e32757043489f69a9c9d90011c501a90230b2aad90b5R43 - I would just reverse the functionality here https://github.com/kinvolk/inspektor-gadget/pull/934/files#diff-398759e651e4fd0d1715f49a7ed93ce34347c1a7086cef5f78761035de23bd42R216 and call it `HasTags()` (and use `!HasTags()` in `WithNoTags()`.) But that's really just my opinion, as I would rather expect `Has` than `HasNo` in the API.

Reply:
`WithNoTags` was introduced in https://github.com/kinvolk/inspektor-gadget/pull/877.